### ISSUE TITLE
Add fields to TABLE_ADDRESS_BOOK: telephone, contact..

### ISSUE THREAD

**Issue by torvista** [zencart/zencart#3253]
I'm thinking of two fields for the address_book table
entry_contact
entry_telephone

Where the installer upgrade will 
copy the customer name (firstname + lastname) -> entry_contact
copy telephone -> entry_telephone

for an imminent PR...

**Comment by scottcwilson** [comment]
This is a great idea.  I don't think it needs to be switchable in admin, just make it global.
I would prefer telephone would be MOVED, not copied.

**Comment by torvista** [comment]
I would not suggest **moving** data, I can hear the screams already/that would break all manner of unknown unknowns.
I just want the fields to be there to be built upon....and also so people can adapt their current/own mods if they are already using such a custom field already.
Duplicating the content was an idea to just to be a bit helpful but with hindsight not a good idea as it will be out of sync with customer data very quickly. Better to let users do that when they are ready.

**Comment by drbyte** [comment]
I'd prefer to exclude the `entry_` prefix.

**Comment by drbyte** [comment]
What's the purpose of adding a "contact name" field, when there's already first-name and last-name?

You'd mentioned that *your* primary goal for this is shipping-related, mainly to add a shipping phone number. 
I gather you're wanting to use "contact" as the "shipping contact name"? Why not actually call it "shipping_contact" so that it suits the actual meaning, and is not conflated with "first-name and last-name"?

But then, no matter what you do, there's crossover about "which" data to use when someone chooses an alternate address-book entry as their "primary". Now which fields are to take priority? Now what's the "semantic" meaning of "contact" compared to first/last or meaning of "telephone" vs previously being "shipping telephone"?

Oy. This just gets very complicated, fast.

**Comment by scottcwilson** [comment]
> I would not suggest **moving** data, I can hear the screams already/that would break all manner of unknown unknowns.
You're right - we just have to be careful to update the phone in the primary address book record any time the customer record is updated.

**Comment by proseLA** [comment]
i'm a fan of adding a phone number for the address book table.  in the previous thread, there was some chatter about not wanting phone numbers on labels which i can only think means a shipping label.  most carriers require a phone number in case they have a problem with delivery.

**Comment by scottcwilson** [comment]
> But then, no matter what you do, there's crossover about "which" data to use when someone chooses an alternate address-book entry as their "primary". Now which fields are to take priority? 

On save, this address's phone overwrites customer.customers_telephone.

**Comment by torvista** [comment]
> What's the purpose of adding a "contact name" field, when there's already first-name and last-name?
My bad, Not needed.

Just for further irritation/example, our courier can use multiple destinations per defined customer (that's our customer, defined in the courier database). These multiple destinations all have telephone and email contacts. The courier sends all the status messages via email, and only uses the telephone when the delivery guy can't get access. They used to send the status by sms, but not anymore. I'm not suggesting another email field, but that's just for information.

So, can we agree on a field name for a telephone field for the address book?

**Comment by DocBenc** [comment]
Also order class should be changed to allow observers in customer, Billing, delivery addresses. Using array and db_perform with core observers could greatly help in mods for further fields without hacking the core code.
Vat, tax code, elettronic invoice code and various additional fields for couriers.
